src/script.c

```diff
@@ -118,18 +129,18 @@ int scriptPrepareForRun(scriptRunCtx *run_ctx, client *engine_client, client *ca
     int obey_client = mustObeyClient(caller);
 
     if (!(script_flags & SCRIPT_FLAG_EVAL_COMPAT_MODE)) {
-        if ((script_flags & SCRIPT_FLAG_NO_CLUSTER) && server.cluster_enabled) {
+        if ((script_flags & SCRIPT_FLAG_NO_CLUSTER) && server.cluster_enabled) { /* unreachable */
             addReplyError(caller, "Can not run script on cluster, 'no-cluster' flag is set.");
             return C_ERR;
         }
 
-        if (!(script_flags & SCRIPT_FLAG_ALLOW_OOM) && server.script_oom && server.maxmemory) {
+        if (!(script_flags & SCRIPT_FLAG_ALLOW_OOM) && server.script_oom && server.maxmemory) { /* unreachable */
```

Comment:
many of the checks and errors here and below are now dead code.
shall we delete them?

Reply:
the dead error checks are these (in non `SCRIPT_FLAG_EVAL_COMPAT_MODE`):
* `MASTERDOWN`
* `Can not run script with write flag on readonly replica`
* `MISCONF`
* `shared.noreplicaserr`
* `OOM`

the live ones are:
* `Can not run script on cluster, 'no-cluster' flag is set.`
* `Can not execute a script with write flag using *_ro command.`
* `shared.masterdownerr` (the one in the `SCRIPT_FLAG_EVAL_COMPAT_MODE` else)

i suppose that if we can't make them reachable, we should either remove them or panic?

Reply:
I vote for panic on the deadcode

Reply:
We realized that some of these are reachable from RM_Call.
Usually, when a module calls a redis command via RM_Call, it skips all the checks (Like OOM) and allows the commands to run anyway, i.e. even write commands despite the OOM state.

but when it calls EVAL, the script engine itself would have always used to do the checks when executing the individual commands the script calls, and i suppose we wanna keep that?
so for scripts with flags, it makes sense that these flags would still be tested before the script starts.

that means that we should either keep these checks in `scriptPrepareForRun`, or we should have RM_Call extract the script flags and act accordingly.
i'll add tests for this and try to decide which of these i prefer.

Reply:
added tests for these in my last commit (misc.tcl)